src/SafeDNS/RecordClient.php

```diff
@@ -55,6 +55,22 @@ public function getZoneRecordById($zoneName, $id)
         return $this->loadEntity($body->data);
     }
 
+    /**
+     * Gets a specific zone record by name
+     *
+     * @param $zoneName
+     * @param string $id
+     * @return Record
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getByName($name)
+    {
+        $response = $this->request("GET", sprintf("v1/records?name=%s", $name));
```

Comment:
This is not valid APIO filtering, see https://developers.ukfast.io/api-guide

Reply:
...also, would this not return a collection instead of a single resource?

Reply:
As this is an index/collection endpoint you may be better implementing a getPage() method like this https://github.com/ukfast/sdk-php/blob/master/src/SafeDNS/ZoneClient.php#L24, then pass in your name filter wherever you are using it.

Reply:
v1/records only returns one record when the exact name is in, similar to a getById, would I not just have to change line 71 to $body->data[0] if it exists?